cmd/minishift/cmd/root.go

```diff
@@ -73,6 +75,13 @@ var RootCmd = &cobra.Command{
 			isAddonInstallRequired bool
 		)
 
+		// If profileName is not an empty string then we need to update
+		// constants.ProfileName to make sure it will create requested directory structure.
+		profileName := isProfileExist(cmd)
+		if profileName != "" {
+			constants.ProfileName = profileName
+		}
```

Comment:
I do not think we need to assign `constants.ProfileName ` again. Because we do it in initconfig() and by the time the control flow reaches here it would have exited in isProfileExist if the profile does not exist.

Reply:
@LalatenduMohanty No, what if profile exist? Then flow will be following.

Assume profile `foo` exist and not as active profile.

```
$ minishift --profile list
foo 
minishift (active)

$ minishift status --profile foo
initconfig() will set the constants.ProfileName to minishift because it's active.
isProfileExist() validate that --profile flag is used so check if valid if yes then return so set back constants.Profile as foo to make sure now all the operation happen on this profile.
```